demo-snippet.html

```diff
@@ -123,6 +150,30 @@
 
         // Stamp the template.
         Polymer.dom(this).appendChild(document.importNode(template.content, true));
+      },
+
+      _copyToClipboard: function() {
+        // From https://github.com/google/material-design-lite/blob/master/docs/_assets/snippets.js
+        var snipRange = document.createRange();
+        snipRange.selectNodeContents(this.$.code);
+        var selection = window.getSelection();
+        selection.removeAllRanges();
+        selection.addRange(snipRange);
+        var result = false;
+        try {
+          result = document.execCommand('copy');
+        } catch (err) {
+          // Copy command is not available
+          console.error(err);
+          this.$.copyButton.icon = 'error';
+        }
+        selection.removeAllRanges();
+        this.$.copyButton.icon = 'done';
```

Comment:
Does it matter that the `icon` is being changed to `'error'` on line 168 above if you are always changing back to `'done'` immediately afterwards?